core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -96,6 +96,11 @@ void* HIPSpace::impl_allocate(const int device_id,
                               const size_t arg_logical_size,
                               [[maybe_unused]] bool stream_sync_only) const {
   void* ptr = nullptr;
+  // ROCm 5.5 and earlier throw an error when using hipMallocAsync and
+  // arg_alloc_size is zero. Instead of trying to allocate memory, just return
+  // early.
+  if (arg_alloc_size == 0) return ptr;
```

Comment:
Do you know why we did only run into this now?

Reply:
No I don't but the problem only appears with `hipMallocAsync` which we set as default last month. So it's not an old piece of code that was intensively tested. I don't know why with multi-gpu we try to do an allocation of size zero.

Reply:
@tcclevenger

Reply:
Please confirm that we don't want to guard this code for older ROCm versions

Reply:
(I saw the description never mind)

Reply:
MultiGPU should not be run for rocm 5.5 or earlier. But I can look into where we would be allocating size 0.

Reply:
Right but these were tests that were only using a single GPU.